nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/PackageReferenceUpdater.cs

```diff
@@ -117,37 +117,34 @@ ILogger logger
         {
             // generate project.assets.json
             var parsedTargetFramework = NuGetFramework.Parse(targetFramework);
-            var tempProject = await MSBuildHelper.CreateTempProjectAsync(tempDir, repoRoot, projectPath, targetFramework, topLevelDependencies, logger, importDependencyTargets: false);
-            var (exitCode, stdOut, stdErr) = await ProcessEx.RunDotnetWithoutMSBuildEnvironmentVariablesAsync(["build", tempProject, "/t:_ReportDependencies"], tempDir.FullName);
+            var tempProject = await MSBuildHelper.CreateTempProjectAsync(tempDir, repoRoot, projectPath, targetFramework, topLevelDependencies, logger, importDependencyTargets: true);
+            var (exitCode, stdOut, stdErr) = await ProcessEx.RunDotnetWithoutMSBuildEnvironmentVariablesAsync(["msbuild", tempProject, "/t:Restore,GenerateBuildDependencyFile"], tempDir.FullName);
             var assetsJsonPath = Path.Join(tempDir.FullName, "obj", "project.assets.json");
             var assetsJsonContent = await File.ReadAllTextAsync(assetsJsonPath);
 
             // build reverse dependency graph
             var assets = JsonDocument.Parse(assetsJsonContent).RootElement;
-            foreach (var tfmObject in assets.GetProperty("targets").EnumerateObject())
+            var tfmObjects = assets.GetProperty("targets").EnumerateObject().ToImmutableArray();
+            if (tfmObjects.Length != 1)
             {
-                var reportedTargetFramework = NuGetFramework.Parse(tfmObject.Name);
-                if (reportedTargetFramework != parsedTargetFramework)
-                {
-                    // not interested in this target framework
-                    continue;
-                }
+                logger.Error($"Expected exactly one target framework group compatible with {targetFramework} but found {tfmObjects.Length}.  Values: {tfmObjects.Select(t => t.Name)}");
+                return (packageParents, packageVersions);
+            }
 
-                foreach (var parentObject in tfmObject.Value.EnumerateObject())
```

Comment:
Nothing below here has changed, just indention.

Reply:
Can we integrate with a linter/formatter to keep this consistent?

Reply:
Makes sense. Thanks for explaining